libflux/src/flux/semantic/flatbuffers/mod.rs

```diff
@@ -1,3 +1,1150 @@
 #![allow(clippy::all)]
 pub mod semantic_generated;
 pub mod types;
+
+use std::cell::RefCell;
+use std::rc::Rc;
+
+use crate::ast;
+
+use crate::semantic;
+use crate::semantic::walk;
+use flatbuffers::{UnionWIPOffset, WIPOffset};
+use semantic_generated::fbsemantic;
+
+extern crate chrono;
+use chrono::Offset;
+
+pub fn serialize(semantic_pkg: &mut semantic::nodes::Package) -> Result<(Vec<u8>, usize), String> {
+    let mut v = new_serializing_visitor_with_capacity(1024);
+    walk::walk(&mut v, Rc::new(walk::Node::Package(semantic_pkg)));
+    v.finish()
+}
+
+fn new_serializing_visitor_with_capacity<'a>(_capacity: usize) -> SerializingVisitor<'a> {
+    SerializingVisitor {
+        inner: Rc::new(RefCell::new(SerializingVisitorState::new_with_capacity(
+            1024,
+        ))),
+    }
+}
+
+struct SerializingVisitor<'a> {
+    inner: Rc<RefCell<SerializingVisitorState<'a>>>,
+}
+
+impl<'a> semantic::walk::Visitor<'_> for SerializingVisitor<'a> {
+    fn visit(&mut self, _node: Rc<walk::Node<'_>>) -> bool {
+        let v = self.inner.borrow();
+        if let Some(_) = &v.err {
+            return false;
+        }
+        Some(SerializingVisitor {
+            inner: Rc::clone(&self.inner),
+        });
+        true
+    }
+
+    fn done(&mut self, node: Rc<walk::Node<'_>>) {
+        let mut v = &mut *self.inner.borrow_mut();
+        if let Some(_) = &v.err {
+            return;
+        }
+        let node = &*node;
```

Comment:
Hmm yeah, I think this was the result of small changes while debugging. I'll try removing it.

Reply:
Okay, so it looks like removing this also removes the reference counter around the node and breaks things a little bit. I'm not super sure why that is, but I'll leave it in for now and try to play around with in in my next PR.

Reply:
I'd be curious to know how it messes with the reference count, mostly out of curiosity with the `Rc` data structure (I've already mentioned my lack of familiarity with it). Does the `&*` signal some sort of op overload that calls `Deref::deref`? What's the error that we get? Maybe there's a more explicit call in `Rc` that might make this operation more explicit? For example, instead of calling `clone` on an `Rc` instance, `Rc::clone` seems preferred, because it makes the reference counting requirement more explicit; maybe there's a better way to express what is required here.

Reply:
@rockstar I think fixing this would be a good opportunity for us to pair together. I'll hit you up and see if we can set up a time.